### ISSUE TITLE
Gmail not registering itself for google cloud messaging!?

### ISSUE THREAD

**Issue by dzizi** [microg/android_packages_apps_GmsCore#398]
I am on LineageOS 14.1 with microG working and push notifications working,
 but cant get gmail to register itself when installing it as new app to receive push notifications, same thing with inbox by google; 
i have 2 registered apps using google cloud messaging (google play store and youtube).

Any tips?

**Comment by codebam** [comment]
Check to make sure that Gmail has the correct app permissions?

**Comment by dzizi** [comment]
It has all permissions.

**Comment by ale5000-git** [comment]
@dzizi: Delete the data of Gmail, it then should re-register itself.

**Comment by dzizi** [comment]
@ale5000-git: I tried but it didn't work. 
Btw i am using aquamail atm; it's working ok and registers itself with gcm.
problem remains...

**Comment by ale5000-git** [comment]
I have tried GMail some time ago and it worked, strange.

Have you tried to completely uninstall GMail (if it is a sytem app also remove the apk in the system partition), restart the phone and then reinstall it?

**Comment by dzizi** [comment]
@ale5000-git: Yes i did try all of that, no luck; (by default it installs in /data/app)

**Comment by gvlfm78** [comment]
@dzizi @ale5000-git Was this issue ever solved? I'm having the same problem

**Comment by ale5000-git** [comment]
I have always installed GmsCore as system app and never had this problem with Gmail.
Maybe it happens when GmsCore isn't a system app.

**Comment by gvlfm78** [comment]
I have it installed in system/priv-app

**Comment by gvlfm78** [comment]
@ale5000-git one further thing I noticed is that if I try to send an email I get a toast notification that GMail cannot connect to Google Play Services and to try again later (I believe sending emails still works though)

**Comment by ale5000-git** [comment]
@gvlfm78: Try to check signature spoofing with [this app](https://f-droid.org/packages/lanchon.sigspoof.checker/).

**Comment by gvlfm78** [comment]
It says signature spoofing is enabled

-------- Original Message --------

**Comment by gvlfm78** [comment]
@ale5000-git the app tells me signature spoofing is enabled

**Comment by ale5000-git** [comment]
Do you have everything checked in microG self check?

**Comment by gvlfm78** [comment]
@ale5000-git Yes, everything is checked

**Comment by gvlfm78** [comment]
@ale5000-git I have just nuked everything related to microG and Gapps and reinstalled microG core + GCM Proxy + UnifiedNlp backends. The issue persists even after reinstalling apps like Gmail and Keep, telling me they are "having trouble with Google Play services. Please try again".

**Comment by ale5000-git** [comment]
@gvlfm78: I'm not sure but you may have missed something if you have done it manually, to to install it with [microG unofficial installer](https://forum.xda-developers.com/showthread.php?t=3432360) to be sure.

**Comment by gvlfm78** [comment]
@ale5000-git Yes that is what I had originally used. After reinstalling it manually 3 times in a row, a few days and reboots later it still not working it randomly started working... I have no idea what made it work...

**Comment by breversa** [comment]
@gvlfm78 You're lucky ! I'm having the same problem as you :
- Lineage for MicroG 14.1 (running on LG G4 H815)
- MicroG v0.2.4-105-gf289a13
- Gmail 8.7.15.206199545.release

... but Gmail doesn't register in my list of GCM apps, though it has all permissions (except calendar, but it shouldn't matter), especially the "Google Mail" extra permission.

**Comment by theel0ja** [comment]
This is not fixing the issue, but I suggest you to use K-9 Mail with Gmail instead.

**Comment by breversa** [comment]
@theel0ja Indeed, it does not fix the issue, as I don't need an _e-mail client for my Gmail account_, but the _Gmail app itself_ to work better/properly. Thanks for your suggestion though, I hope it'll help someone. :-)

**Comment by gvlfm78** [comment]
Well, now Gmail doesn't seem to show any Gmail accounts as logged in despite them being so. It even has all permissions.

**Comment by breversa** [comment]
Has anyone found a way to make it work yet ?
